Sherwin-77/go-tix
# Feat/entity repository

**Summary**
This pull request introduces an entity-repository layer to streamline database operations and improve separation of concerns in the project.

**Changes Made**
- Created a new repository layer for managing Entity database operations.
- Added interface definitions for the repository to enable easy mocking and testing.
- Implemented CRUD methods in the repository:
  - CreateEntity: Adds a new entity to the database.
  - GetEntityByID: Retrieves an entity by its ID.
  - GetAllEntities: Fetches all entities from the database.
  - UpdateEntity: Updates an existing entity.
  - DeleteEntity: Deletes an entity by its ID.
- Updated service and handler layers to use the repository for database interactions.

**How to Test**

1. Run Unit Tests:

- Execute the test suite for the repository layer:
`go test ./repository -v`

- Verify all tests pass, and edge cases are handled.

2. Integration Testing:

- Run the application and test the endpoints that interact with the repository.
- Ensure CRUD operations work as expected.

**Checklist**

- [ &#x2705]  Repository layer implemented with all CRUD methods.
- [&#x2705 ]  Documentation updated (if applicable).
